File: Softwaresystems/Python AlgoDat/Algorithms/Sorting/bubble_sort.py
def bubble_sort(arr):
    n = len(arr)

    for i in range(n-1):
        for j in range(0, n-i-1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]

arr = [12,45,78,98,63,21,458,78,96,420,69,21,78,96,55,266,666,214,789,654,1,2,4,8,9,3,5,78]

bubble_sort(arr)
print("Sorted array is: ")
print(arr)

File: Softwaresystems/Informatik2/FS23/00 Sorting/bubbleSort.c
#include <stdio.h>

void printer(int arr[], int n){
    for(int i = 0; i < n; i++){
        printf("%d ", arr[i]);
    }
}

void swap(int *a, int *b) {
  int temp = *a;
  *a = *b;
  *b = temp;
}

void bubbleSort(int arr[], int n){
    for(int i = 0; i < n - 1; i++){
        for(int j = 0; j < n - 1 - i; j++){
            if(arr[j] > arr[j + 1]){

                //swapping using pointers
                swap(&arr[j], &arr[j + 1]);

                //alternative way wihtout the usage of pointers
                //int temp = arr[j];
                //arr[j] = arr[j+1];
                //arr[j+1] = temp;
            }
        }
    }
}

int main(){
    int arr[5] = {3, 7, 1, 12, 9};
    bubbleSort(arr, 5);

    printer(arr, 5);

}